vignettes/update.Rmd

```diff
@@ -0,0 +1,254 @@
+
+Since adding/removing/updating frosting as well as adjusting a layer in a `frosting` object proceeds in the same way as performing those tasks on an `epi_recipe`, we will focus on implementing those for an `epi_recipe` in this vignette and only briefly go through some examples for a `frosting` object.
+
+## Add/update/remove an `epi_recipe` in an `epi_workflow`
+
+We start with the built-in `case_death_rate_subset` dataset that contains JHU daily COVID-19 cases and deaths by state and take a subset of it from Nov. 1, 2021 to Dec. 31, 2021 for the four states of Alaska, California, New York, and South Carolina.
+
+```{r}
+jhu <- case_death_rate_subset %>%
+  dplyr::filter(time_value >= as.Date("2021-11-01"), geo_value %in% c("ak", "ca", "ny", "sc"))
+
+jhu
+```
+
+Then, we construct a simple `epi_recipe` named `r`, where we lag the death rates by 0, 7, and 14 days, lead the death rate by 14 days, omit NA values in all predictors and then in all outcomes (and set `skip = TRUE` to skip over this processing of the outcome variable when the recipe is baked). 
+
+```{r}
+r <- epi_recipe(jhu) %>%
+  step_epi_lag(death_rate, lag = c(0, 7, 14)) %>%
+  step_epi_ahead(death_rate, ahead = 14) %>%
+  step_naomit(all_predictors()) %>%
+  step_naomit(all_outcomes(), skip = TRUE)
+```
+
+We add this recipe to an `epi_workflow` object by inputting `r` into the `add_epi_recipe` function:
+
+```{r}
+wf <- epi_workflow() %>%
+  add_epi_recipe(r)
+
+wf
+```
+
+We may then go on to add the fitted linear model to our `epi_workflow`:
+```{r}
+# Fit a linear model
+wf <- epi_workflow(r, parsnip::linear_reg()) %>% fit(jhu)
+
+wf
+```
+
+At this stage, suppose we decide to overhaul our recipe so that we have a different set of pre-processing steps or we want to make multiple changes to existing steps, but we desire to keep the remainder of the `epi_workflow` the same. We can use the `update_epi_recipe` function to trade our current recipe `r` for another recipe `r2` in `wf` as follows:
+
+```{r}
+r2 <- epi_recipe(jhu) %>%
+  step_epi_lag(death_rate, lag = c(0, 1, 7, 14)) %>%
+  step_epi_lag(case_rate, lag = c(0:7, 14)) %>%
+  step_epi_ahead(death_rate, ahead = 7) %>%
+  step_epi_naomit()
+
+wf <- update_epi_recipe(wf, r2)
+wf
+```
+You can see that the output of `wf` depicts the sequence of steps in `r2` instead of `r`, which indicates that the update was successful.
+
+A longer approach to achieve the same end is to use `remove_epi_recipe` to remove the old recipe and then `add_epi_recipe` to add the new one. Under the hood, the `update_epi_recipe` function operates in this way. 
+
+The `add_epi_recipe` and `remove_epi_recipe` functions offload to the `workflows` versions of the functions (`workflows::add_recipe()` and `workflows::remove_recipe()`) as much as possible. The main reason for using the `epipredict` version is so that we ensure that we do not lose the `epi_workflow` class. 
```

Comment:
I prefer package names to go in curly braces. (Good use of paren on the function names here.)

```suggestion
The `add_epi_recipe()` and `remove_epi_recipe()` functions offload to the `{workflows}` versions of the 
functions  as much as possible. The main reason for using the `{epipredict}` version is so that we ensure 
that we retain the `epi_workflow` class. 
```